src/db/highlights.rs

```diff
@@ -0,0 +1,60 @@
+use super::*;
+use itertools::Itertools;
+
+impl Db {
+    pub async fn get_highlights(&self) -> Result<HashMap<String, Vec<UserId>>> {
+        let mut cache = self.highlight_cache.lock().await;
+        // we have a cache
+        if let Some(cache) = cache.as_ref() {
+            return Ok(cache.clone());
+        // we don't have a cache at all
+        } else {
+            let mut conn = self.pool.acquire().await?;
+            let q = sqlx::query!("select * from highlights")
+                .fetch_all(&mut conn)
+                .await?
+                .into_iter()
+                .map(|x| (x.word, UserId::from(x.user as u64)))
+                .into_group_map();
+            cache.replace(q.clone());
+            return Ok(q.clone());
```

Comment:
no need for the `return` keyword here.
also, do you need to clone here? is q not owned?